File: 14. Longest Common Prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
  const len = strs.length;
  if (len === 0) return "";

  let prefix = strs[0];
  for (let i = 1; i < len; i++) {
    while (strs[i].indexOf(prefix) !== 0) {
      prefix = prefix.substring(0, prefix.length - 1);
      if (prefix === "") return "";
    }
  }

  return prefix;
};

File: 14. Longest Common Prefix/solution.go
func longestCommonPrefix(strs []string) string {

	if len(strs) == 0 {
		return ""
	}

	prefix := strs[0]
	for i := 1; i < len(strs); i++ {
		for strings.HasPrefix(strs[i], prefix) == false {
			prefix = string([]rune(prefix)[:len(prefix) - 1])
		}
	}

	return prefix
}